addons/mass_mailing/static/src/js/mailing_mailing_view_form_full_width.js

```diff
@@ -151,6 +172,14 @@ const MassMailingFullWidthFormController = FormController.extend({
             });
         }
     },
+
+    /**
+     * The themes have been loaded by the mass mailing widget.
+     */
+    _onThemesLoaded() {
```

Comment:
That's ... a bit hackish but quite short. Maybe asking @Zinston what he thinks of this feature and its addition to the mailing editor integration ? In short: allowing to filter out template bodies (aka "favorite mailing.mailing records") based on target model.

Reply:
This seems like a good idea to me because I don't think you would want to use the same template for a mass mailing as for a notification email for instance. But this is a question for @LNAodoo.
As for the implementation, it's hacky but pretty in line with the way things are done in the front end in general :eyes: 
I do wonder though, isn't it possible to do that filtering in the python code and via a param (eg, `model_id`) when fetching the themes by rpc ?

Reply:
I might have misunderstood the context here then. Is there any way to change the model without starting a new editor ? Because if not, then you are refetching in any case.

Reply:
@Zinston when you change the model, the editor is not updated, (so we fetch the templates only once)

The other solution would be to trigger an event from the form view, and listen this event in the mailing widget... But again, the widget will only work in this view which is weird

Reply:
Ah right gotcha. This feels like it would be even hackier for no real gain so I guess your original option is best.

Reply:
I also think, it's way more simple (and also when you change the model, the template are displayed instantly, with the other solution you will get a little "lag")

Reply:
Be aware though that the file you put it in is really only used when in the main form view of mass_mailing so not for instance in Marketing Automation.

Reply:
@Zinston in the mailing form view, you can select the model, so inside the same view the model can change (it's just a many2one field, aka mailing list VS mailing contact VS lead,...)

Reply:
@Zinston 
cc @tde-banana-odoo 
to have this behavior in marketing automation, I'll need to create a new file (and probably inherit from this new class in "mailing full width) just for that (so the diff will be bigger), do you think it's worth it ?

Reply:
or maybe I can add an option on the widget to fetch only the templates of 1 model, so
- in marketing automation we set this option as we can not change the model
- in mass mailing we do not set this option and fetch all templates (and they are hidden by the form view code)

What do you think ?

Reply:
@Zinston 
> are you talking about the Recipients selector?

Yes, this is this field :)